diff -r 1099346a20d5 -r be390a9828b7 libavcodec/dv_profile.c
--- a/libavcodec/dv_profile.c	Tue Jul 18 16:03:47 2017 +0200
+++ b/libavcodec/dv_profile.c	Tue Jul 18 16:03:47 2017 +0200
@@ -165,7 +165,7 @@
       .ltc_divisor        = 30,
       .height             = 1080,
       .width              = 1280,
-      .sar                = { { 1, 1 }, { 3, 2 } },
+      .sar                = { { 3, 2 }, { 3, 2 } },
       .pix_fmt            = AV_PIX_FMT_YUV422P,
       .bpm                = 8,
       .block_sizes        = block_sizes_dv100,
@@ -182,7 +182,7 @@
       .ltc_divisor        = 25,
       .height             = 1080,
       .width              = 1440,
-      .sar                = { { 1, 1 }, { 4, 3 } },
+      .sar                = { { 4, 3 }, { 4, 3 } },
       .pix_fmt            = AV_PIX_FMT_YUV422P,
       .bpm                = 8,
       .block_sizes        = block_sizes_dv100,
@@ -199,7 +199,7 @@
       .ltc_divisor        = 60,
       .height             = 720,
       .width              = 960,
-      .sar                = { { 1, 1 }, { 4, 3 } },
+      .sar                = { { 4, 3 }, { 4, 3 } },
       .pix_fmt            = AV_PIX_FMT_YUV422P,
       .bpm                = 8,
       .block_sizes        = block_sizes_dv100,
@@ -216,7 +216,7 @@
       .ltc_divisor        = 50,
       .height             = 720,
       .width              = 960,
-      .sar                = { { 1, 1 }, { 4, 3 } },
+      .sar                = { { 4, 3 }, { 4, 3 } },
       .pix_fmt            = AV_PIX_FMT_YUV422P,
       .bpm                = 8,
       .block_sizes        = block_sizes_dv100,
