diff -r 222bcecacc24 -r 132b9c57f021 configure
--- a/configure	Tue Sep 08 11:22:35 2015 +0200
+++ b/configure	Tue Sep 08 11:22:59 2015 +0200
@@ -4534,7 +4534,6 @@
 die_license_disabled gpl libsmbclient
 die_license_disabled gpl libutvideo
 die_license_disabled gpl libvidstab
-die_license_disabled gpl libx264
 die_license_disabled gpl libx265
 die_license_disabled gpl libxavs
 die_license_disabled gpl libxvid
