diff -r be390a9828b7 -r f86f04198fcd libavformat/rtpenc.c
--- a/libavformat/rtpenc.c	Tue Jul 18 16:03:47 2017 +0200
+++ b/libavformat/rtpenc.c	Tue Jul 18 16:03:47 2017 +0200
@@ -33,6 +33,7 @@
     { "payload_type", "Specify RTP payload type", offsetof(RTPMuxContext, payload_type), AV_OPT_TYPE_INT, {.i64 = -1 }, -1, 127, AV_OPT_FLAG_ENCODING_PARAM },
     { "ssrc", "Stream identifier", offsetof(RTPMuxContext, ssrc), AV_OPT_TYPE_INT, { .i64 = 0 }, INT_MIN, INT_MAX, AV_OPT_FLAG_ENCODING_PARAM },
     { "cname", "CNAME to include in RTCP SR packets", offsetof(RTPMuxContext, cname), AV_OPT_TYPE_STRING, { .str = NULL }, 0, 0, AV_OPT_FLAG_ENCODING_PARAM },
+    { "timestamp", "Starting timestamp", offsetof(RTPMuxContext, base_timestamp), AV_OPT_TYPE_INT,  { .i64 = -1 }, INT_MIN, INT_MAX, AV_OPT_FLAG_ENCODING_PARAM },
     { "seq", "Starting sequence number", offsetof(RTPMuxContext, seq), AV_OPT_TYPE_INT, { .i64 = -1 }, -1, 65535, AV_OPT_FLAG_ENCODING_PARAM },
     { NULL },
 };
@@ -114,7 +115,8 @@
         st->id = s->payload_type;
     }
 
-    s->base_timestamp = av_get_random_seed();
+    if (s->base_timestamp == -1)
+      s->base_timestamp = av_get_random_seed();
     s->timestamp = s->base_timestamp;
     s->cur_timestamp = 0;
     if (!s->ssrc)
