diff -r b244b3c23743 -r 50cd330f1558 libavformat/movenc.c
--- a/libavformat/movenc.c	Tue Jul 18 16:08:07 2017 +0200
+++ b/libavformat/movenc.c	Tue Jul 18 16:08:07 2017 +0200
@@ -1844,7 +1844,7 @@
     if (track->st)
         t = av_dict_get(track->st->metadata, "reel_name", NULL, 0);
 
-    if (t && utf8len(t->value))
+    if (t && utf8len(t->value) && track->mode != MODE_MP4)
         mov_write_source_reference_tag(pb, track, t->value);
     else
         avio_wb16(pb, 0); /* zero size */
@@ -2221,7 +2221,10 @@
     } else if (track->tag == MKTAG('r','t','p',' ')) {
         mov_write_hmhd_tag(pb);
     } else if (track->tag == MKTAG('t','m','c','d')) {
-        mov_write_gmhd_tag(pb, track);
+        if (track->mode == MODE_MP4)
+            mov_write_nmhd_tag(pb);
+        else
+            mov_write_gmhd_tag(pb, track);
     }
     if (track->mode == MODE_MOV) /* FIXME: Why do it for MODE_MOV only ? */
         mov_write_hdlr_tag(pb, NULL);
@@ -5072,7 +5075,7 @@
         }
     }
 
-    if (mov->mode == MODE_MOV) {
+    if (mov->mode == MODE_MOV || mov->mode == MODE_MP4) {
         tmcd_track = mov->nb_streams;
 
         /* +1 tmcd track for each video stream with a timecode */
